kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+	}
+
+	private List<Payee> removePayeesBelowThreshold(List<Payee> payees, Integer year) {
+		List<Payee> retval = new ArrayList<Payee>();
+		Map<Integer, Map<String, Double>> totals = new HashMap<Integer, Map<String, Double>>();
+		Map<String, KualiDecimal> taxAmountByPaymentType = getTaxAmountByPaymentType(true);
+		Map<String, Double> payeeTotalsMap;
+
+		// loop over payees and rollup amount to parent
+		for (Payee p : payees) {
+			payeeTotalsMap = totals.get(p.getVendorHeaderGeneratedIdentifier());
+
+			if (payeeTotalsMap == null) {
+				totals.put(p.getVendorHeaderGeneratedIdentifier(), payeeTotalsMap = new HashMap<String, Double>());
+			}
+
+			// sum up amounts by different types
+			for (Payment payment : (List<Payment>) p.getPayments()) {
+				if (!payment.getExcludeIndicator() && year.equals(p.getTaxYear())) {
+					if (payment.getAcctNetAmount().isNonZero()) {
+						Double taxAmount = payeeTotalsMap.get(payment.getPaymentTypeCode());
+						if (taxAmount == null) {
+							payeeTotalsMap.put(payment.getPaymentTypeCode(), taxAmount = Double.valueOf(0));
+						}
+
+						taxAmount = Double.valueOf(taxAmount.doubleValue() + payment.getAcctNetAmount().doubleValue());
+						payeeTotalsMap.put(payment.getPaymentTypeCode(), taxAmount);
+						totals.put(p.getVendorHeaderGeneratedIdentifier(), payeeTotalsMap);
+					}
+				}
+			}
+		}
+
+		double taxThreshhold = getTaxThreshold(true);
+
+		// loop over payees and add any that meet minimum required amount based on type
+		for (Payee p : payees) {
+			payeeTotalsMap = totals.get(p.getVendorHeaderGeneratedIdentifier());
+
+			if (payeeTotalsMap != null) {
+				for (String paymentType : payeeTotalsMap.keySet()) {
+					Double amount = payeeTotalsMap.get(paymentType);
+
+					KualiDecimal d = taxAmountByPaymentType.get(paymentType);
+					// if this limit is setup in parameters
+					if ((d != null) && (amount.doubleValue() >= d.doubleValue())) {
+						retval.add(p);
+						break;
+					} 
+					else if (amount.doubleValue() >= taxThreshhold) { // default limit
+						retval.add(p);
+						break;
+					}
+				}
+			}
+		}
+		return retval;
+	}
+
+	private double getTaxThreshold(boolean forceReload) {
+		if (forceReload || (taxThreshold == 0)) {
+			taxThreshold = TaxHelper.getIncomeThreshold(parameterService);
+		}
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("taxThreshhold: " + taxThreshold);
+		}
+
+		return taxThreshold;
+	}
+
+	private Map<String, KualiDecimal> getTaxAmountByPaymentType(boolean forceReload) {
+		if (forceReload || (taxAmountByPaymentType == null)) {
+			taxAmountByPaymentType = TaxHelper.getTaxAmountByPaymentType(parameterService);
+		}
+
+		if (LOG.isDebugEnabled()) {
+			if (taxAmountByPaymentType != null) {
+				for (String s : taxAmountByPaymentType.keySet()) {
+					LOG.debug(s + "=" + taxAmountByPaymentType.get(s));
+				}
+			}
+		}
+
+		return taxAmountByPaymentType;
+	}
+
+	public List<String> getObjectCodes(List<String> levels, String type) {
```

Comment:
Please keep as is. This is an artifact of utilizing a repository design pattern that is non-generic. To genericize would be too big of a refactor.